src/ExpressionEvaluator/CSharp/Source/ExpressionCompiler/CompilationContext.cs

```diff
@@ -1270,6 +1271,11 @@ private static DkmClrCompilationResultFlags GetLocalResultFlags(LocalSymbol loca
                     var instance = new DisplayClassInstanceFromParameter(parameter);
                     displayClassInstances.Add(new DisplayClassInstanceAndFields(instance));
                 }
+                else
+                {
+                    // Collect parameter names that will hide display class variables from outer scopes
+                    userDefinedParameterNames.Add(parameter.Name);
```

Comment:
>userDefinedParameterNames.Add(parameter.Name); [](start = 20, length = 46)

Is there a reason to do this for a top level method?

Reply:
There is no outer scope at the top level, so no filtering will take place. I don't know to tell whether we're dealing with a top-level method without inspecting the parameters anyways.

---
In reply to: [574721854](https://github.com/dotnet/roslyn/pull/51129#discussion_r574721854) [](ancestors = 574721854)